File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
  int height;

  do {
    printf("Enter a height (1-23): \n");
    height = GetInt();
  } while (height < 0 || height > 23);

  if (height == 0) {
    return 0;
  }

  for (int i = 0; i < height; i++) {
    for (int j = 0; j < height-(i+1); j++) {
      printf("%s", " ");
    }
    for (int k = 0; k < i+2; k++) {
      printf("#");
    }
    printf("\n");
  }
}

File: pset1/mario.pl
use strict;
use warnings;

sub main {

  my $height;

  do {
    print "Enter a height (1-23): \n";
    $height = <STDIN>;
    chomp $height;
  } while ($height =~ /\D/ || $height < 0 || $height > 23);

  if ($height == 0) {
    print "0\n";
  }

  for (my $i = 0; $i < $height; $i++) {
    for (my $j = 0; $j < $height-($i+1); $j++) {
      print " ";
    }
    for (my $k = 0; $k < $i+2; $k++) {
      print "#";
    }
    print "\n";
  }
}

main();
